pages/EntanglmentState.py

```diff
@@ -0,0 +1,203 @@
+    result = job.result()
+    counts = result.get_counts()
+    job_monitor(job)
+    job.status()
+    return counts
+
+# ########## Streamlit Interface #############
+n = st.slider("Select number of qubits", 2, 10, 3)
+entanglement_scheme = st.selectbox("Select the entanglement scheme", ["GHZ State", "W State"])
+
+providers = {"Basic Aer": BasicAer, "Aer": Aer}
+selected_provider = st.selectbox("Select Provider", list(providers.keys()))
+backends = providers[selected_provider].backends()
+selected_backend = st.selectbox("Select Backend", [backend.name() for backend in backends])
+
+if st.button("Generate State"):
+    if entanglement_scheme == "GHZ State":
+        circuit = create_ghz_state(n)
+    else:
+        circuit = create_w_state(n)
+
+    st.write("**Circuit**")
+    fig, ax = plt.subplots()
+    circuit_drawer(circuit, output='mpl', ax=ax)  # Display the circuit using Matplotlib
+    st.pyplot(fig)  # Show the Matplotlib figure in Streamlit
+
+    counts = run_circuit(circuit, providers[selected_provider], selected_backend)
+    st.subheader("Result Counts:")
+    st.write(counts)  # Display result counts as text
+    st.subheader("Histogram:")
+    fig, ax = plt.subplots()  # Create a Matplotlib figure and axes
+    plot_histogram(counts, ax=ax)  # Pass the axes to the plot_histogram function
+    st.pyplot(fig)  # Display the Matplotlib figure using st.pyplot
+
+# ################################### Show the Implementation #########################################
+
+st.subheader("Implementation of Entanglement State")
+bv_algo_code = """
+
+# ###############Importing libraries###################
+from qiskit import QuantumCircuit, BasicAer, Aer, execute
+from qiskit.visualization import plot_histogram
+from qiskit.tools import job_monitor
+from qiskit import IBMQ
+from random import choice
+
+# ########## GHZ State Circuit ################
+def create_ghz_state(n):
+    ghz_circuit = QuantumCircuit(n, n)
+    ghz_circuit.h(0)
+    for qubit in range(1, n):
+        ghz_circuit.cx(0, qubit)
+    ghz_circuit.barrier()
+    for i in range(n):
+        ghz_circuit.measure(i, i)
+    return ghz_circuit
+
+# ########## W State Circuit ##################
+def create_w_state(n):
+    w_circuit = QuantumCircuit(n, n)
+    w_circuit.ry(2 * acos(sqrt(1/n)), 0)
+    for qubit in range(1, n):
+        w_circuit.cx(qubit - 1, qubit)
+        w_circuit.ry(-2 * acos(sqrt(1/(n - qubit))), qubit)
+    w_circuit.barrier()
+    for i in range(n):
+        w_circuit.measure(i, i)
+    return w_circuit
+
+def run_circuit(circuit, provider, backend_name, shots=1024):
+    backend = provider.get_backend(backend_name)
+    job = execute(circuit, backend=backend, shots=shots)
+    result = job.result()
+    counts = result.get_counts()
+    job_monitor(job)
+    job.status()
+    return counts
+
+"""
+
+
+st.code(bv_algo_code, language="python")
+
+st.markdown("""
+# Quantum Entanglement Visualization: GHZ and W States
```

Comment:
Use `st.subheader` instead of `st.markdown`. In this section, you don't need to add an explanation of the code. Instead, you have to add an explanation of what is 'GHZ and W States' are, how to create to, the quantum circuit for it, and mathematics for them. (From mathematics, I mean how the state evolves from the application of each gate in the circuit. I will attach a sample reference to it.).